Classes/CZ805M.uc

```diff
@@ -2,11 +2,323 @@ class CZ805M extends MP7MMedicGun;
 
 var int MaxHealAmmo;
 
+var MedicDartAttachment FakedHealDartAttach; //faked healing dart attachment
+
 var         name             ReloadShortAnim;
 var         float             ReloadShortRate;
 
 var transient bool  bShortReload;
 
+simulated function PostBeginPlay()
+{
+    super.PostBeginPlay();
+    
+    if ( Level.NetMode != NM_DedicatedServer ) {
+        if ( FakedHealDartAttach == none )
+        FakedHealDartAttach = spawn(class'MedicDartAttachment',self);
+        if ( FakedHealDartAttach != none ) {
+            FakedHealDartAttach.SetDrawScale(4.0); //scale
+            AttachToBone(FakedHealDartAttach, 'body'); //attach to the CZ805M main bone, 'body'
+			FakedHealDartAttach.SetRelativeRotation(rot(-250,-16834,0)); //pitch yaw roll
+			FakedHealDartAttach.SetRelativeLocation(vect(1.14, 9.5, 1.13)); //x y z
+        }
+    }
+}
+
+//destroy it 
+simulated function Destroyed()
+{
+    if ( FakedHealDartAttach != None )
+        FakedHealDartAttach.Destroy();
+    super.Destroyed();
+}
+
+//hides mag1 bullets when firing, bullets 1 to 10 are in the magwell so you'll never see them though, bullets 11 to 30 are scaled according to magazine size
+simulated function Notify_HideBullets()
+{
+    local float MagFloat; 
+    local int MyAmmos;
+    
+    //don't do any of this stuff on servers
+    if ( (Level.NetMode == NM_DedicatedServer) || (AIController(Instigator.Controller) != None) )
+		return;
+    
+    MyAmmos = MagAmmoRemaining;
+    if (MyAmmos > 9)
+    {
+        MagFloat = (MyAmmos-10)/(MagCapacity-10);
+    }
+    if (MyAmmos < 1)
```

Comment:
So many ifs here. How about creating an array of struct with ammo and bone name fields. Then you could simply iterate through the array.

Reply:
OK, I'll refactor this myself, so you'll have a source for future copypasting :smile: